src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -45,5 +59,94 @@
     }
   }
 
+  /**
+   * Get the historical price data for the passed in symbol over the passed in range.
+   *
+   * @param symbol stock symbols to get historical data for.
+   * @param range selection of time for which to get historical data.
+   * @return a list of historical price objects for the Symbol that is passed in covering the date
+   *     range.
+   */
+  public List<IexHistoricalPrice> getHistoricalPricesForSymbol(final String symbol,
+      final String range) {
+    if ("".equals(symbol) || "".equals(range)) {
+      return Collections.emptyList();
+    } else {
+
+      boolean queryInDB = true;
+      LocalDate today = LocalDate.now();
+      LocalDate initialDate = calculateStartingDate(today, range);
+      List<LocalDate> datesOverRange = initialDate.datesUntil(today).collect(Collectors.toList());
+      List<IexHistoricalPrice> pricesOverRange = new ArrayList<>();
+      for (LocalDate date: datesOverRange) {
+        if(holidays.contains(date) || date.getDayOfWeek() == DayOfWeek.SATURDAY || date.getDayOfWeek() == DayOfWeek.SUNDAY){
+          continue;
+        }
+        Optional<IexHistoricalPrice> datePrice = historicalRspy.findById(symbol.toUpperCase() + date);
+        if(datePrice.isEmpty()){
+          queryInDB = false;
+          break;
+        }else{
+          pricesOverRange.add(datePrice.get());
+        }
+      }
+
+      if(queryInDB){
+        log.info("Elements for symbol: "+symbol+", range: "+range+" retrieved from local storage.");
+        return pricesOverRange;
+      }else{
+        List<IexHistoricalPrice> pricesForSymbol = iexNewClient.getHistoricalPricesForSymbol(symbol, range, TOKEN);
+        int dataCount = 0;
+        for(LocalDate date : datesOverRange){
+          if(holidays.contains(date) || date.getDayOfWeek() == DayOfWeek.SATURDAY || date.getDayOfWeek() == DayOfWeek.SUNDAY){
+            continue;
+          }
+          IexHistoricalPrice historicalPrice = pricesForSymbol.get(dataCount);
+          if(!date.toString().equals(historicalPrice.getDate())){
+            holidays.add(date);
+          }else{
+            dataCount++;
+            historicalPrice.setSymbolAndDate(historicalPrice.getSymbol()+historicalPrice.getDate());
+            historicalRspy.save(historicalPrice);
+          }
+        }
+        log.info("Elements for symbol: "+symbol+", range: "+range+" retrieved via IEX request, then stored locally.");
+        return pricesForSymbol;
+      }
+    }
+  }
+
+  /**
+   * Calculates the starting date for traversing over the range given. Handles integer-day/month/year increments,
+   *
+   * @param today current date.
+   * @param range length of time into the past in integer+day/month/year format.
+   * @return starting date or null if cannot be calculated from given range.
+   */
+  private LocalDate calculateStartingDate(LocalDate today, String range){
+    if(range.equals("ytd")){
```

Comment:
```ytd```, ```m```, etc are all "magic strings." I happen to personally know what they mean because I've done this before, but it's good practice to give string constants names, and extract them to ```private static final String```s at the top of the class.
e.g. ```private static final String MONTH_RANGE_INDICATOR = "m"```